src/app/clusters/operational-state-server/operational-state-server.cpp

```diff
@@ -0,0 +1,151 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/util/af.h>
+
+#include <app/util/attribute-storage.h>
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/callback.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/enums.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app/CommandHandler.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/ConcreteCommandPath.h>
+#include <app/util/error-mapping.h>
+#include <lib/core/CHIPEncoding.h>
+
+using namespace chip;
+using namespace chip::app::Clusters::OperationalState;
+using namespace chip::app::Clusters::OperationalState::Attributes;
+
+void emberAfOperationalStateClusterInitCallback(chip::EndpointId endpoint)
+{
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "--------> %s()", __FUNCTION__);
+    return;
+}
+
+void emberAfOperationalStateClusterServerInitCallback(chip::EndpointId endpoint)
+{
+    // TODO
+    // Get from the "real thermostat"
```

Comment:
What does this have to do with thermostats?

I strongly suggest breaking out the XML additions into a separate PR, and doing implementation in a PR on top of that.

Reply:
I will remove them